hugegraph-core/src/main/java/com/baidu/hugegraph/task/StandardTaskScheduler.java

```diff
@@ -248,127 +272,128 @@ private void unlistenChanges() {
+                /*
+                 * Remove task from memory if it's running on worker node,
+                 * but keep task in memory if it's running on master node.
+                 * cancel-scheduling will read task from backend store, if
+                 * removed this instance from memory, there will be two task
+                 * instances with same id, and can't cancel the real task that
+                 * is running but removed from memory.
+                 */
+                this.remove(task);
+            }
             // Notify master server to schedule and execute immediately
             TaskManager.instance().notifyNewTask(task);
+            return;
         }
+
+        throw new HugeException("Can't cancel task '%s' in status %s",
+                                task.id(), task.status());
+    }
+
+    protected ServerInfoManager serverManager() {
+        return this.serverManager;
     }
 
-    protected synchronized <V> void scheduleTasks() {
-        // Master schedule all queued tasks to suitable servers
+    protected synchronized void scheduleTasks() {
+        // Master server schedule all queued tasks to suitable worker servers
+        Map<HugeServerInfo, MutableInt> scheduleInfos = new HashMap<>();
+        ServerInfoManager serverManager = this.serverManager();
         String page = this.supportsPaging() ? PageInfo.PAGE_NONE : null;
         do {
-            Iterator<HugeTask<V>> tasks = this.tasks(TaskStatus.QUEUED,
-                                                     PAGE_SIZE, page);
-            HugeTask<V> task;
+            Iterator<HugeTask<Object>> tasks = this.tasks(TaskStatus.SCHEDULING,
+                                                          PAGE_SIZE, page);
             while (tasks.hasNext()) {
-                task = tasks.next();
+                HugeTask<?> task = tasks.next();
                 if (task.server() != null) {
                     // Skip if already scheduled
                     continue;
                 }
-                HugeServerInfo server = this.pickWorker(task);
+
+                HugeServerInfo server = serverManager.pickWorker(task);
                 if (server == null) {
                     LOG.debug("The master can't find suitable servers to " +
-                              "execute task: {}, wait for next schedule",
+                              "execute task '{}', wait for next schedule",
                               task.id());
-                    return;
+                    continue;
                 }
 
                 // Found suitable server, update task server and server load
                 assert server.id() != null;
                 task.server(server.id());
+                task.status(TaskStatus.SCHEDULED);
                 this.save(task);
+
                 server.load(server.load() + task.load());
-                this.serverManager().save(server);
-                LOG.info("Schedule task {} to server {}",
+
```

Comment:
must save server info here. Otherwise all tasks will be scheduled to same worker

Implement `increaseLoad(int load)` in class `HugeServerInfo`

Reply:
use batch read write